python/lsst/daf/butler/datastores/fileLikeDatastore.py

```diff
@@ -595,6 +639,69 @@ def getUri(self, ref, predict=False):
 
         return location.uri
 
+    @transactional
+    def trash(self, ref):
+        """Indicate to the Datastore that a Dataset can be removed.
```

Comment:
FWIW, I have mostly been replacing "Dataset" with "dataset" when I see it in docs, since it's not actually a class.  Happy to stop doing that, but you and I should probably be consistent.